ostd/src/arch/x86/cpu/mod.rs

```diff
@@ -385,96 +388,166 @@ cpu_context_impl_getter_setter!(
     [gsbase, set_gsbase]
 );
 
-/// The floating-point state of CPU.
+pub(super) static CPU_FEATURES: Once<FeatureInfo> = Once::new();
+pub(super) static XSTATE_MAX_FEATURES: Once<u64> = Once::new();
+pub(super) static XSAVE_SIZE: Once<usize> = Once::new();
```

Comment:
Do not organize Rust code in the C style. Rust code can use items defined later in the text. So put high-level, public, and important concepts such as `FpRegs` at the beginning and the low-level, less important, private stuffs at the end. Code should be read (and written) like a newspaper, with the most important and interesting stuff at the top of the page.